File: RAI/EjemploBanco/JavaImplementation/Cliente.java
public class Cliente{
    private String rut;
    private String nombre;
    private Producto[] misProductos;
    private int limite;
    private int cupo;

    public Cliente(String rut, String nombre){
        this.rut = rut;
        this.nombre = nombre;
        this.misProductos = new Producto[4];
        this.limite = 4;
        this.cupo = 4;
    } 

    public String getNombre(){
        return this.nombre;
    }

    public void addProducto(Producto nuevo){
        if(this.cupo > 0){
            this.misProductos[this.limite - this.cupo] = nuevo;
            this.cupo--;
        }
    }

    public String toString(){
        String productos = "";
        for(Producto p:this.misProductos)
            productos += "\n" + p.toString();
        return "Cliente (" + this.rut + "): " + this.nombre + "\nProductos:" + productos;
    }

}

File: RAI/EjemploBanco/python_implementation/Cliente.py
from Producto import Producto

class Cliente:
    def __init__(self, rut, nombre):
        self.rut = rut
        self.nombre = nombre
        self.mis_productos = [None] * 4
        self.limite = 4
        self.cupo = 4

    def add_producto(self, nuevo: Producto):
        if self.cupo > 0:
            self.mis_productos[self.limite - self.cupo] = nuevo
            self.cupo -= 1

    def __str__(self):
        str_productos = ""
        for p in self.mis_productos:
            str_productos += "\n" + str(p)

        return "Cliente (" + self.rut + "): " + self.nombre + "\nProductos:" + str_productos